cmd/installer/root.go

```diff
@@ -1219,6 +1224,74 @@ type PacksInstallationType struct {
 	PackIdx string
 }
 
+type updateCfg struct {
+	Default struct {
+		Date string
+		Auto bool
+	}
+}
+
+func (p *PacksInstallationType) checkUpdateCfg(v *viper.Viper, conf *updateCfg) error {
+	v.SetConfigFile(filepath.Join(p.WebDir, "update.cfg"))
+	v.SetConfigType("ini")
+	if err := v.ReadInConfig(); err != nil {
+		return err
+	}
+	if err := v.Unmarshal(conf); err != nil {
+		return err
+	}
+	if t, err := time.Parse("2-1-2006", conf.Default.Date); err != nil {
+		return err
+	} else {
+		if time.Since(t).Hours() > 24 { // index.pidx older than 1 day
+			return errs.ErrIndexTooOld
+		}
+	}
+	return nil
+}
+
+func (p *PacksInstallationType) updateUpdateCfg(v *viper.Viper, conf *updateCfg) error {
+	_ = v
```

Comment:
Left out?
```suggestion
```

Please also remove all commented-out code, unless needed, in such case please add the reason.

Reply:
Then I need to save that version of this file. Viper does not work well if writing a file, maybe I've not understood the non existant documentation. If I have time for that then it should be in again.

Reply:
The _ = v is to suppress the warning for unused parameters. If viper is used again, then this parameter will be needed.